rclcpp/src/rclcpp/parameter_client.cpp

```diff
@@ -337,6 +372,22 @@ SyncParametersClient::has_parameter(const std::string & parameter_name)
   return vars.names.size() > 0;
 }
 
+std::vector<rcl_interfaces::msg::ParameterDescriptor>
+SyncParametersClient::describe_parameters(const std::vector<std::string> & parameter_names)
+{
+  auto f = async_parameters_client_->describe_parameters(parameter_names);
+
+  using rclcpp::executors::spin_node_until_future_complete;
+  if (
+    spin_node_until_future_complete(
+      *executor_, node_base_interface_,
+      f) == rclcpp::FutureReturnCode::SUCCESS)
+  {
+    return f.get();
+  }
```

Comment:
This is purely subjective, but I think this is kind of ugly.  What if instead we did:

```
rclcpp::FutureReturnCode future = spin_node_until_future_complete(*executor_, node_base_interface_, f);
if (future == rclcpp::FutureReturnCode::SUCCESS) {
  return f.get();
}
```

Reply:
agree with you.